domain/port/service/service.go

```diff
@@ -224,6 +229,9 @@ func (s *Service) UpdateUnitPorts(ctx context.Context, unitUUID coreunit.UUID, o
 					openPorts[endpoint] = append(openPorts[endpoint][:i], openPorts[endpoint][i+1:]...)
 				}
 			}
+			if len(openPorts[endpoint]) == 0 {
+				delete(openPorts, endpoint)
+			}
```

Comment:
I get why we do this, but it does seem like a code smell.

Reply:
Agree, I'm not entirely happy. Tbh the service layer here to too complex, it's working with too many layers of abstraction.

It's due a re-write when we remove RunAtomic, so I've been putting off a tidy which will fix these smells

Reply:
Maybe I should just bite the bullet and do it now, but I'm concerned that work will be wasted